src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,134 @@
+```
+
+Using [`cmake`](https://cmake.org/) and a C++ compiler with C++11 or later, build the custom TensorFlow op.
+
+```sh
+mkdir build
+cd build
+cmake ..
+make
+```
+
+A yaml file must be created with options for training. An example of this is in [the example_net.yaml file](https://github.com/Fastcode/VisualMesh/blob/master/example_net.yaml).
+
+Execute the training by running
+
+```sh
+./mesh.py train [config.yaml] [output_dir]
+```
+
+Where `[config.yaml]` is the options yaml file as specified above, and the output directory is the name of the directory you want the progress and results to be saved to. Progress can be monitored with TensorBoard.
+
+### Testing
+
+Testing the network is executed similarly to training. Run
+
+```sh
+./mesh.py test [config.yaml] [output_dir]
+```
+
+Where `[config.yaml]` and `[output_dir]` are the same as for training. The output will be the average precision for each class, as well as the precision/recall curves and the final mean average precision.
+
+### Exporting Configuration
+
+The weights and biases will be output into a yaml file like the one found at [model.yaml on the Visual Mesh repository](https://github.com/Fastcode/VisualMesh/blob/master/example/model.yaml) or [VisualMesh.yaml on the NUbots repository](https://github.com/NUbots/NUbots/blob/master/module/vision/VisualMesh/data/config/VisualMesh.yaml).
+
+Replace the NUbots repository `VisualMesh.yaml` configuration with the configuration you just obtained from training the Visual Mesh. Ensure you create a new branch so you can make a PR and get any changes merged in. For further information about Git, view the [Git Guide](/guides/general/git).
+
+## Camera Calibration
+
+If the cameras are not calibrated correctly, the vision system will not work well. Information on calibration is available on [the input page](/system/subsystems/input).
+
+An automatic camera calibration tool is available to use in the NUbots repository that can be used to calibrate the cameras. A guide on how to calibrate the cameras will be available on NUbook soon.
+
+## Testing
+
+Now that the Visual Mesh has been updated in the NUbots repository, you should test it before merging.
+
+Build the code, ensuring `VisualMesh.role` is on in `./b configure -i`. Go to the [Getting Started page](/guides/main/getting-started) to find out more on how to build the code and install onto a robot. Ensure the new configuration file has been installed. Check out the [Build System page](/system/foundations/build-system) to find out more about options when installing onto the robot.
+
+When your new Visual Mesh is installed onto the robot, connect to the robot and ensure NUsight is on.
+
+```sh
+nano config/NUsight.yaml
+```
+
+Turn `vision object` and `compressed images` on. Run NUsight using `yarn prod` and navigate to the NUsight page in your browser. More on NUsight can be found on [the NUsight page](/system/tools/nusight).
+
+Run the visualmesh role
+
+```sh
+./visualmesh
```

Comment:
wouldn't you do `./b run visualmesh`?

Reply:
If you're running on the robot it'll just be `./visualmesh`.

Reply:
Oh, I missed the part where it said connect to the robot :sweat_smile: